src/tree_store/page_store/mmap.rs

```diff
@@ -1,23 +1,43 @@
 use crate::Result;
+use errno::errno;
+use memmap2::MmapRaw;
+use std::fs::File;
+use std::io;
 use std::ops::Range;
+use std::os::unix::io::AsRawFd;
 use std::slice;
 
 pub(crate) struct Mmap {
+    file: File,
     inner: memmap2::MmapRaw,
 }
 
 impl Mmap {
-    pub(crate) fn new(mmap: memmap2::MmapRaw) -> Self {
-        Self { inner: mmap }
+    pub(crate) fn new(file: File) -> Result<Self> {
+        Ok(Self {
+            inner: MmapRaw::map_raw(&file)?,
+            file,
+        })
     }
 
     pub(crate) fn len(&self) -> usize {
         self.inner.len()
     }
 
+    #[cfg(target_os = "macos")]
     pub(crate) fn flush(&self) -> Result {
-        self.inner.flush()?;
+        // TODO: It may be unsafe to mix F_BARRIERFSYNC with writes to the mmap.
+        //       Investigate switching to `write()`
+        let code = unsafe { libc::fcntl(self.file.as_raw_fd(), libc::F_BARRIERFSYNC) };
+        if code == -1 {
+            return Err(io::Error::new(io::ErrorKind::Other, errno()).into());
```

Comment:
Would `Error::last_os_err()` do the same thing? I'd prefer to avoid the dependency on the `errno` crate, if possible

Reply:
Sweet that looks like it'll do. Fixed it.